### PR TITLE
LPS-177330 Add SelectEventEntityInput component to handle entities for events

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
This PR contains is a continuation of [this](https://github.com/liferay-tango/liferay-portal/pull/3269) one. It adds the capability to use the item selector for Downloaded Documents and Media events.
[Link to the task](https://issues.liferay.com/browse/LPS-177007)
[Link to the story](https://issues.liferay.com/browse/LPS-173020)

Proposed Solution
========
The solution consist in adding a new component to render the Entity Input for events because It differs from the other entity inputs used until now. Moreover, the intention is to be extend to be used with the rest of events. Right now its showing the fileEntryId which is going to be changed in the near future.

Steps to Verify:
----------------
1. Connect to AC to DXP.
2. Go to AC -> Segments -> Dynamic Segment -> Events -> Downloaded Documents and Media and create a segment.
3. Go to Applications Menu -> Control Menu -> Job Scheduler -> Scheduled Jobs -> (page 2) -> Click on Run Now for the CheckIndividualSegmentsMessageListener to sync up the segments between AC and DXP 
4.  Go to ProductMenuPanel -> People -> Segments
5. Edit the segment created in AC by clicking on it and clicking on the toggle button
6. Open a modal with the DXP documents and select a new one.
7. Check that the input from the Entity changes to the one selected.

### CODE REVIEW

**Review by claraizquierdo — APPROVED** [review]
Hi @Stephy6 , everything seems to be working fine 👌 

I just deleted what I think that was an unnecessary property.

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by claraizquierdo** [comment]
Hello again @Stephy6 @cgoncas ,
I added a commit transforming the selected document ID to sha256hex. If you approve we can send this PR :)

**Comment by claraizquierdo** [comment]
ci:test:sf

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js [hunk 1]
```diff
@@ -51,7 +52,10 @@ class SelectEventEntityInput extends React.Component {
 				const value = JSON.parse(event.value);
 
 				onChange({
-					assetId: value.fileEntryId,
+					assetId: hash
+						.sha256()
+						.update(value.fileEntryId)
+						.digest('hex'),
```
Comment (p2kmgcl) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js:58]: Do we really need to use `hash.js`? We can create the same assetId using the native API, it is well supported. This is from MDN (https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#converting_a_digest_to_a_hex_string):

```js
async function digestMessage(message) {
  const msgUint8 = new TextEncoder().encode(message); // encode as (utf-8) Uint8Array
  const hashBuffer = await crypto.subtle.digest("SHA-256", msgUint8); // hash the message
  const hashArray = Array.from(new Uint8Array(hashBuffer)); // convert buffer to byte array
  const hashHex = hashArray
    .map((b) => b.toString(16).padStart(2, "0"))
    .join(""); // convert bytes to hex string
  return hashHex;
}
```

**Review by p2kmgcl — COMMENTED** [review]
Hey @Stephy6! It looks great :smile:, I just added a comment to check if we could avoid using `hash.js`, as it is only being used in `analytics-client-js` and we can probably remove that.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js (hunk 1 above)
Comment (Stephy6) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js:58]: Oh I'll take a look, thank you @p2kmgcl !

**Review by cgoncas — APPROVED** [review]
Everything looks good from my point of view.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEventEntityInput.es.js]: Updated! :smile:

**Comment by Stephy6** [comment]
ci:test:relevant

**Review by claraizquierdo — APPROVED** [review]
LGTM!!! Thank you @p2kmgcl for the improvement!

**Review by p2kmgcl — APPROVED** [review]

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by p2kmgcl** [comment]
ci:reevaluate:1267401_8091

**Comment by p2kmgcl** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
@CarlosBrichete @rafaellajordao , could you confirm that the failing tests are not related to the PR?

**Comment by Stephy6** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:reevaluate:1267401_8104

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by CarlosBrichete** [comment]
Hey guys, this test LocalFile.ACSegmentsInDXP#RedirectToSegmentPageAC is not failing in CI but it has failed because of the same reason in the past, so I think it is a little bit flaky. I think we can assume that the test failure is not related to this PR, so we can do a forward-force :)

**Comment by Stephy6** [comment]
ci:test:relevant

**Comment by Stephy6** [comment]
ci:forward:force

**Comment by Stephy6** [comment]
ci:forward:force

**Pull request closed without merging**